Changed URL schema. Modified @format decorator

rezniknn opened the pull request.

rezniknn committed:
added configuration file to store tornado options - options.conf. Refactored app.py to use options from config file

rezniknn committed:
removed duplicated import

rezniknn committed:
Changed URL schema. Modified @format decorator

mansuleman commented:
Based on the API document SCPI command passed through the modules handler, i.e. /modules/?serial=ID&scpi=SCPI_COMMAND
Please make sure that this commit corresponds to the API document https://docs.google.com/document/d/17jka7XUzQt5636tGPPW6-f6fDyBZjouqWoKAbQNsXdE/edit#

user2589 commented on easy_phi/app.py:
```diff
@@ -4,7 +4,7 @@
 """
 import json
-
+import dicttoxml
```
it is not in default lib, need to add to requirements.txt

user2589 commented on easy_phi/app.py:
```diff
@@ -124,11 +126,11 @@ def get(self):
-    (r"/modules", ModulesListHandler),
-    (r"/pick_module", SelectModuleHandler),
-    (r"/send_command", SCPICommandHandler),
+    (r"/", PlatformInfoHandler),  # Default page
```
missing api version info

user2589 replied:
default page is web interface, not an API call. Here it is handled by the same view as api call in next line

mansuleman replied:
It is better to implement PICK modules through modules handler by specifying particular port ID. Same with send_command. Please refer to document for that  https://docs.google.com/document/d/17jka7XUzQt5636tGPPW6-f6fDyBZjouqWoKAbQNsXdE/edit#

rezniknn replied:
I don't agree. Pick module and Display list of modules are two semantically different operations. First assumes modifying the internal state of the program (locking the module to a particular user), therefore should be implemented using POST request. The second one is a simple query of the internal state of the program, therefore should be implemented using GET request. Since the two operations have nothing in common I suggest to implement them using two separate RequestHandlers

rezniknn replied:
Doc needs to be update after review is accepted

rezniknn replied:
Same goes for your first comment above

user2589 commented on easy_phi/app.py:
```diff
@@ -124,11 +126,11 @@ def get(self):
-    (r"/send_command", SCPICommandHandler),
+    (r"/", PlatformInfoHandler),  # Default page
+    (r"/api/info", PlatformInfoHandler),
+    (r"/api/modules", ModulesListHandler),
```
missing api version info, e.g. /api/v1/modules

user2589 committed:
Merge branch 'dev' into alexey_131_config_file

user2589 committed:
added usb ports configuration

degradation: now code uses synchronous read from modules

user2589 committed:
removed debug output

user2589 committed:
Merge branch 'alexey_131_config_file' into alexey_132_url_schema

Conflicts:
	easy_phi/app.py

user2589 committed:
fix after review

- added dicttoxml to requirements
- added api version into URLs
- debug mode xml prettyprint
- changed content type
- added JSONP support

mansuleman commented on easy_phi/app.py:
```diff
@@ -123,16 +157,26 @@ def get(self):
+    (r"/api/v1/info", PlatformInfoHandler),
+    (r"/api/v1/modules", ModulesListHandler),
+    (r"/api/v1/module", SelectModuleHandler),
+    (r"/api/v1/scpi", SCPICommandHandler),
```
Module handler can still do both GET and POST at the same time, not an issue. 
GET - to select module
POST - to send SCPI command to the module

Please see example
 http://stackoverflow.com/questions/24899201/is-it-possible-to-use-the-same-method-for-both-get-and-post-with-webapp2

Let's see what @user2589 thinks

user2589 replied:
I agree with @rezniknn that at this point we don't have time to pass all design decisions through official document. 
Now, about design decisions. GET requests should not change anything in the system, and you effectively propose to set status (selected/not selected) by GET request. Also, for user selection we need both GET and POST to read and update status.
As for SCPI commands, we need to get module list of available commands (aka configuration), which does not change module status, and send SCPI commands which may or may not change something in the system. So, we need both GET and PUT  there as well.
Also, please note that user lock and SCPI concepts are semantically two different entities, I would prefer not to mix them in one handler

rezniknn committed:
Dynamic api version

mansuleman commented:
ship it

user2589 committed:
moved decorators to separate module

user2589 committed:
Merge branch 'alexey_132_url_schema' of github.com:QuantumCMU/easy_phi_API into alexey_132_url_schema

Conflicts:
	easy_phi/app.py

user2589 committed:
added welcome message to platform info

user2589 commented:
to be fixed: 
- remove @api_handler decorator
- options.default_format is not used
